src/main/java/graphql/normalized/ExecutableNormalizedOperationFactory.java

```diff
@@ -453,6 +453,15 @@ private Set<GraphQLObjectType> narrowDownPossibleObjects(Set<GraphQLObjectType>
         if (currentOnes.isEmpty()) {
             return resolvedTypeCondition;
         }
+
+        // Intersection calculation is expensive when either set is large.
+        // If a set only has one member, it is equivalent and cheaper to calculate via contains.
+        if (resolvedTypeCondition.size() == 1 && currentOnes.contains(resolvedTypeCondition.iterator().next())) {
+            return resolvedTypeCondition;
+        } else if (currentOnes.size() == 1 && resolvedTypeCondition.contains(currentOnes.iterator().next())) {
+            return currentOnes;
+        }
```

Comment:
Java shits me on how access to a first member is that god awful `collection.iterator().next()`

But here we are!

Reply:
Perhaps create a utitlity method in `graphql.util.FpKit`

`FpKit. intersection(` that does this optimisation for same type sets

Reply:
That's a good idea I'll stick it in `FpKit` instead